src/richie/apps/core/context_processors.py

```diff
@@ -86,3 +92,93 @@ def site_metas(request):
 
     context["FRONTEND_CONTEXT"] = json.dumps(context["FRONTEND_CONTEXT"])
     return context
+
+
+class WebAnalyticsContextProcessor:
+    """
+    Context processor to add Web Analytics tracking information to Richie CMS templates and
+    frontend.
+    """
+
+    def context_processor(self, request: HttpRequest) -> dict:
+        """
+        Real implementation of the context processor for the Web Analytics core app sub-module
+        """
+        context = {}
+        if hasattr(request, "current_page"):
+            # load web analytics settings to the context
+            if getattr(settings, "WEB_ANALYTICS_ID", None):
+                context["WEB_ANALYTICS_ID"] = settings.WEB_ANALYTICS_ID
+            if getattr(settings, "WEB_ANALYTICS_LOCATION", None):
+                context["WEB_ANALYTICS_LOCATION"] = settings.WEB_ANALYTICS_LOCATION
+            if getattr(settings, "WEB_ANALYTICS_PROVIDER", None):
+                context["WEB_ANALYTICS_PROVIDER"] = settings.WEB_ANALYTICS_PROVIDER
+            if self.is_load_dimensions(request):
+                context["WEB_ANALYTICS_DIMENSIONS"] = self.get_dimensions(request)
```

Comment:
Make it simpler?
```suggestion
            context["WEB_ANALYTICS_DIMENSIONS"] = self.get_dimensions(request) if getattr(settings, "WEB_ANALYTICS_ID", None) else None
```

Reply:
Well, that depends the opinion of each developer. Personally I don't like to abuse on the inline if's that is formatted to multi-line if's. Nevertheless, I've changed with you proposition the result is:
```python
           context["WEB_ANALYTICS_DIMENSIONS"] = (
                self.get_dimensions(request)
                if getattr(settings, "WEB_ANALYTICS_ID", None)
                else None
            )
```